src/openms/source/QC/TopNoverRT.cpp

```diff
@@ -164,4 +168,29 @@ namespace OpenMS
   {
     return QCBase::Status() | QCBase::Requires::RAWMZML | QCBase::Requires::POSTFDRFEAT;
   }
+
+  // get maximal intensity
+  float TopNoverRT::getBasePeakIntensity_(const MSSpectrum& spec)
+  {
+    float peak_max{0};
+    for (const Peak1D& peak : spec)
```

Comment:
since we're iterating over the spectrum twice now, its probably better to merge the two functions and compute two values within one loop (performance)?!

Reply:
just pass in two IntensityTypes and return void